tests/unit/_internal/runner/test_strategy.py

```diff
@@ -0,0 +1,31 @@
+import os
+import typing as t
+
+import bentoml
+from bentoml._internal.runner import strategy
+from bentoml._internal.resource import get_resource as resource_get_resource
+from bentoml._internal.runner.strategy import DefaultStrategy
+
+
+class GPURunnable(bentoml.Runnable):
+    SUPPORTED_RESOURCES = ("nvidia.com/gpu",)
+
+
+def unvalidated_get_resource(x: t.Dict[str, t.Any], y: str):
+    return resource_get_resource(x, y, validate=False)
+
+
+def test_default_gpu_strategy(monkeypatch):
+    monkeypatch.setattr(strategy, "get_resource", unvalidated_get_resource)
+    assert DefaultStrategy.get_worker_count(GPURunnable, {"nvidia.com/gpu": 2}) == 2
+    assert DefaultStrategy.get_worker_count(GPURunnable, {"nvidia.com/gpu": 0}) == 1
```

Comment:
it was working as expected @sauyon . Now if resources are not specified it will default to a worker count of 1 and give a warning saying "no resource found, falling back to using a single worker" which was the logic earlier

Reply:
Yeah it is the behavior caused by login in SchedulingStrategy. I wonder if it's correct but we may solve that in another PR